### PR TITLE
Button component should be disabled while loading is true

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Button still clickable while loading is true

#### What problem is this solving?
Now the user should wait for the loading be false to click on the button and trigger the action again.

#### How should this be manually tested?
[workspace](https://felipe--storecomponents.myvtex.com/)

#### Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by kevinch** [comment]
I don't agree. This is something that the developer using the Button should implement on the app side with the use of the `disabled` prop.

File: react/components/Button/index.js [hunk 1]
```diff
@@ -1,23 +1,29 @@
-import React, { Component } from 'react'
+import React, { Component, Fragment } from 'react'
 import PropTypes from 'prop-types'
 
 import Spinner from '../Spinner'
 
 class Button extends Component {
+  state = {
+    shouldBeDisabled: false,
```
Comment (kevinch) [inline react/components/Button/index.js]: Disagree, this adds up way too much logic for a component that should be _dumb_.

**Comment by klzns** [comment]
Ta muito mais complexo esse state management do que uma variavel, que nao tem custo nenhum.

E no handleClick seria o mesmo if:

```js
!this.props.isDisabled && !this.props.isLoading && this.props.onClick && this.props.onClick(event)
```

E @kevinch como todo mundo vai ter que fazer esse comportamento no loading, é mais facil o componente ter isso na logica dele do que o pai. Acho que vc concorda comigo tambem, so a solucao usando state que talvez tenha causado essa reação.

File: react/components/Button/index.js [hunk 2]
```diff
@@ -1,23 +1,19 @@
...
 import PropTypes from 'prop-types'
 
 import Spinner from '../Spinner'
 
 class Button extends Component {
   handleClick = event => {
-    !this.props.disabled && this.props.onClick && this.props.onClick(event)
-  };
+    !this.props.disabled && !this.props.isLoading
+    this.props.onClick && this.props.onClick(event)
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply react/components/Button/index.js]: 👍

**Pull request merged**